<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <group ns="/rms">

        <!-- Robot model -->
        <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find rms_scara)/urdf/scara0.xacro'"/>
        
        <!-- Gazebo Spawn -->
        <arg name="x" default="0"/>
        <arg name="y" default="0"/>
        <arg name="z" default="0.2"/>
     
        <node name="mybot_spawn" pkg="gazebo_ros" type="spawn_model" output="screen"
              args="-urdf -param robot_description -model scara0 -x $(arg x) -y $(arg y) -z $(arg z)" />

        <!-- Load Controller -->
        <rosparam command="load" file="$(find rms_scara)/config/scara0.yaml"/>

        <!-- Controler -->
        <node name="controller_spawner" pkg="controller_manager" type="spawner"
            respawn="false" output="screen" ns="/rms"
            args="--namespace=/rms
            joint_state_controller
            joint1_position_controller
            joint2_position_controller
            joint3_position_controller
            gripper_right_position_controller
            gripper_left_position_controller
            --timeout 60">
        </node>

    </group>

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find rms_scara)/worlds/pen_blue_world.world"/>
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="true"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
    </include>

    <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
        <remap from="image" to="/rms/camera1/image_raw" />
    </node>
 
</launch>
